content/en/docs/tasks/configure-pod-container/resize-container-resources.md

```diff
@@ -2,264 +2,192 @@
+* `RestartContainer`: Restart the container to apply the new resource values. This is often necessary for memory changes because many applications and runtimes cannot adjust their memory allocation dynamically.
+
+If `resizePolicy[*].restartPolicy` is not specified for a resource, it defaults to `NotRequired`.
+
 {{< note >}}
-In the above example, if desired requests or limits for both CPU _and_ memory
-have changed, the container will be restarted in order to resize its memory.
+If a Pod's overall `restartPolicy` is `Never`, then any container `resizePolicy` must be `NotRequired` for all resources. You cannot configure a resize policy that would require a restart in such Pods.
 {{< /note >}}
 
-<!-- steps -->
+**Example Scenario:**
+
+Consider a container configured with `restartPolicy: NotRequired` for CPU and `restartPolicy: RestartContainer` for memory.
+* If only CPU resources are changed, the container is resized in-place.
+* If only memory resources are changed, the container is restarted.
+* If *both* CPU and memory resources are changed simultaneously, the container is restarted (due to the memory policy).
 
 ## Limitations
 
-In-place resize of pod resources currently has the following limitations:
+In-place resource resizing currently has the following limitations:
+
+* **Resource Types:** Only CPU and memory resources can be resized.
+* **Memory Decrease:** Memory limits _cannot be decreased_ unless the `resizePolicy` for memory is `RestartContainer`. Memory requests can generally be decreased.
+* **QoS Class:** The Pod's original [Quality of Service (QoS) class](/docs/tasks/configure-pod-container/quality-service-pod/) (Guaranteed, Burstable, BestEffort) is determined at creation and cannot be changed by a resize. The resized resource values must still adhere to the rules of the original QoS class:
+    * *Guaranteed*: Requests must continue to equal limits for both CPU and memory after resizing.
+    * *Burstable*: Requests and limits cannot become equal for *both* CPU and memory simultaneously (as this would change it to Guaranteed).
+    * *BestEffort*: Resource requirements (`requests` or `limits`) cannot be added (as this would change it to Burstable or Guaranteed).
+* **Container Types:** Non-restartable Init Containers and Ephemeral Containers cannot be resized. [Sidecar containers](/docs/concepts/workloads/pods/sidecar-containers/) can be resized.
+* **Resource Removal:** Resource requests and limits cannot be entirely removed once set; they can only be changed to different values.
+* **Operating System:** Windows pods do not support in-place resize.
+* **Node Policies:** Pods managed by [static CPU or Memory manager policies](/docs/tasks/administer-cluster/cpu-management-policies/) cannot be resized in-place.
+* **Swap:** Pods utilizing [swap memory](/docs/concepts/architecture/nodes/#swap-memory) cannot resize memory requests unless the `resizePolicy` for memory is `RestartContainer`.
```

Comment:
Is this true?
```suggestion
* **Swap:** Pods utilizing [swap memory](/docs/concepts/architecture/nodes/#swap-memory) can only resize memory requests if the `resizePolicy` for memory is `RestartContainer`. If the `resizePolicy` is set to something else, the API server accepts the resize request but it will not succeed.
```

Reply:
Again, do you think it's necessary to describe the enforcement mechanism here? Here's the full list:

Enforced in API admission:
- Resource type
- QoS class
- Container types
- Windows pods (only if OS name is windows)
- Resource removal

Enforced at allocation time (resize marked Infeasible):
- Static CPU / memory
- Swap
- node Capacity (not listed here)
- Windows (anything that slipped through admission)